### PR TITLE
[m3admin] Add Replace to placement client

### PR DESCRIPTION
**Pull request by schallert**

### CODE REVIEW

File: pkg/m3admin/placement/types.go [hunk 1]
```diff
@@ -38,4 +38,6 @@ type Client interface {
 	Add(instance placementpb.Instance) error
 	// Remove removes a given instance with the given ID from the placement.
 	Remove(id string) error
+	// Replace replaces one instance with another.
+	Replace(leavingID string, newInstance placementpb.Instance) error
```
Comment (CK-Ward) [inline pkg/m3admin/placement/types.go]: i don't know how much i like the name `leavingID` but i don't have any better ideas lol

File: pkg/controller/m3admin_client.go [hunk 2]
```diff
@@ -221,3 +221,7 @@ func (c errorPlacementClient) Add(instance placementpb.Instance) error {
 func (c errorPlacementClient) Remove(id string) error {
 	return c.err
 }
+
+func (c errorPlacementClient) Replace(_ string, _ placementpb.Instance) error {
```
Comment (CK-Ward) [inline pkg/controller/m3admin_client.go]: why do these parameters have underscores?

File: pkg/m3admin/placement/types.go (hunk 1 above)
Comment (schallert) [reply pkg/m3admin/placement/types.go]: Ah ya know what, I'm going to change it to `leavingInstanceID` to match the placement service [API](https://godoc.org/github.com/m3db/m3cluster/placement#Service) and the [proto](https://github.com/m3db/m3/blob/85841a3ff2b11b69549e6ae2f19cfa750408c850/src/query/generated/proto/admin/placement.proto#L25)

File: pkg/controller/m3admin_client.go (hunk 2 above)
Comment (schallert) [reply pkg/controller/m3admin_client.go]: It's a thing I've seen in Go before to indicate the parameter isn't used in the function, but I just realized I can drop the `_` entirely! Going to do that

**Pull request merged**